src/ioctls/device.rs

```diff
@@ -80,6 +80,104 @@ impl DeviceFd {
+    ///                   (i.e. the `addr` field) to access the returned device attribute
+    ///                   data.
+    ///
+    /// # Returns
+    ///
+    /// * Returns the last occured `errno` wrapped in an `Err`.
+    ///
+    /// * `device_attr` - The `addr` field of the `device_attr` structure will point to
+    ///                   the device attribute data.
+    ///
+    /// # Examples
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// # extern crate kvm_bindings;
+    /// # use kvm_ioctls::Kvm;
+    ///
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    ///
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// {
+    ///     use kvm_bindings::{
+    ///         KVM_DEV_VFIO_GROUP, KVM_DEV_VFIO_GROUP_ADD,
+    ///         kvm_device_type_KVM_DEV_TYPE_VFIO,
+    ///     };
+    ///
+    ///     // Create a VFIO device.
+    ///     let mut vfio_device = kvm_bindings::kvm_create_device {
+    ///         type_: kvm_device_type_KVM_DEV_TYPE_VFIO,
+    ///         fd: 0,
+    ///         flags: 0,
+    ///     };
+    ///
+    ///     let device_fd = vm
+    ///         .create_device(&mut vfio_device)
+    ///         .expect("Cannot create KVM VFIO device");
+    ///
+    ///     let mut dist_attr_mut = kvm_bindings::kvm_device_attr {
+    ///         group: KVM_DEV_VFIO_GROUP,
+    ///         attr: u64::from(KVM_DEV_VFIO_GROUP_ADD),
+    ///         addr: 0x0,
+    ///         flags: 0,
+    ///     };
+    ///
+    ///     // On x86_64, getting device attributes from `KVM_DEV_VFIO_GROUP`
+    ///     // will cause an error, as nothing can be extracted.
+    ///     assert!(device_fd.get_device_attr(&mut dist_attr_mut).is_err());
+    /// }
+    ///
+    /// #[cfg(any(target_arch = "aarch64"))]
+    /// {
+    ///     use kvm_bindings::{
+    ///         KVM_DEV_ARM_VGIC_GRP_NR_IRQS, kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2,
+    ///         kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3,
+    ///     };
+    ///
+    ///     // Create a GIC device.
+    ///     let mut gic_device = kvm_bindings::kvm_create_device {
+    ///         type_: kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3,
+    ///         fd: 0,
+    ///         flags: 0,
+    ///     };
+    ///     let device_fd = match vm.create_device(&mut gic_device) {
+    ///        Ok(fd) => fd,
+    ///        Err(_) => {
+    ///        gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2;
+    ///        vm.create_device(&mut gic_device)
+    ///            .expect("Cannot create KVM vGIC device")
+    ///        }
+    ///     };
+    ///
+    ///     let mut data: u32 = 0;
+    ///     let mut gic_attr_get = kvm_bindings::kvm_device_attr::default();
+    ///     gic_attr_get.group = KVM_DEV_ARM_VGIC_GRP_NR_IRQS;
+    ///     gic_attr_get.addr = &mut data as *const u32 as u64;
+    ///
+    ///     assert!(device_fd.get_device_attr(&mut gic_attr_get).is_ok());
```

Comment:
Can you please replace the assert with `unwrap`? Asserts obfuscate the code a bit and the code does not flow naturally anymore.

Reply:
Done